File: c++/problem-23(Two_Dishes)/TWODISH.cpp
#include <iostream>
using namespace std;

int main() {
	// your code goes here
	int t;
	cin>>t;
	while(t--){
	    int N,A,B,C;
	    cin>>N>>A>>B>>C;
	    if (B <N){
	         cout<<"NO"<<endl;
	    }
	    else{
	        if ((A+C) >= N){
	            cout<<"YES"<<endl;
	        }
	        else{
	            cout<<"NO"<<endl;
	        }
	    }
	}
	return 0;
}

File: python/problem-23(Two_Dishes)/TWODISH.py
# cook your dish here
t = int(input())
for i in range(t):
    N,A,B,C = map(int,input().split())
    total = A+C
    if (B<N):
        print('NO')
    else:
        if total >= N:
            print('YES')
        else:
            print('NO')
    